package NYT.NHiveClient.NProto;

import "yt/core/misc/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqCommitTransaction
{
    required NYT.NProto.TGuid transaction_id = 1;
    repeated NYT.NProto.TGuid participant_cell_ids = 2;
}

message TRspCommitTransaction
{
    required int64 commit_timestamp = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqAbortTransaction
{
    required NYT.NProto.TGuid transaction_id = 1;
    optional bool force = 2 [default = false];
}

message TRspAbortTransaction
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqPingTransaction
{
    required NYT.NProto.TGuid transaction_id = 1;
    // Only makes sense for master transactions.
    optional bool ping_ancestors = 2 [default = false];
}

message TRspPingTransaction
{ }

////////////////////////////////////////////////////////////////////////////////
